src/rosdep2/main.py

```diff
@@ -918,6 +935,127 @@ def command_resolve(args, options, lookup=None):
         return 1  # error exit code
 
 
+def search_cached_data_source(view, regexes):
+    def count_errors(results):
+        if FALL_BACK_TO_RE:
+            return 0
+        return sum([sum(r.fuzzy_counts) for r in results])
+
+    close_keys = []
+    close_pkgs = []
+    for key in view.rosdep_data:
+        results = [regex.search(key) for regex in regexes]
+        if all(results):
+            error = count_errors(results)
+            close_keys.append({'key': key, 'error': error})
+            continue
+        # If key is no match, check if any of the packages are a match
+        item = view.rosdep_data[key]
+        for os_entry in item:
+            os_items = item[os_entry]
+
+            def check_pkgs(items):
+                for pkg in items:
+                    results = [regex.search(pkg) for regex in regexes]
+                    if all(results):
+                        error = count_errors(results)
+                        close_pkgs.append({'key': key, 'pkg': pkg, 'os': os_entry, 'error': error})
```

Comment:
I'm a bit confused by the intended or unintended use of the `os` info for packages:

```
$ rosdep search tracetools --rosdistro=rolling
WARNING: ROS_PYTHON_VERSION is unset. Defaulting to 3
Closest keys:
  tracetools_image_pipeline
  tracetools
  tracetools_launch
  tracetools_read
  tracetools_test
  tracetools_trace
  tracetools_acceleration
  tracetools_analysis

Closest packages:
  ros2trace_analysis: ros/rolling/tracetools_analysis [osx]

```

Why is it only listing `osx`? As a user, I would think this means that this package is only available on macOS. Could we instead list all OSes, e.g., `[osx, debian, rhel, ubuntu]`? The code seems to be stopping after finding results for the first available OS.

Reply:
I've decided to stop when a match is found because usually it's not relevant if there's another os with a matching name, since you usually know what you are looking for.
The main use case here is also looking for the package name in your distro's package manager to find the associated key, and in that case, only that would match anyway.
Debian and Ubuntu, for example, would almost always both have the same results, and this would unnecessarily bloat the output.
If I were to also list the other OS, it would be confusing if I didn't also list the matching package name for each OS, and this could get quite a long list if the search string matches many packages.

Reply:
What about searching just for the currently used OS (or the one passed in `--os`) ?

Reply:
Yeah, I think that would make a lot of sense. Not sure what the proper way to do it is, but looks like you could use `OsDetect` + the OS override info.

> I've decided to stop when a match is found because usually it's not relevant if there's another os with a matching name, since you usually know what you are looking for.

If the OS information doesn't really matter, it shouldn't be shown IMO.

Reply:
Okay, I have changed it to use the os information and only look through packages for the current OS.
This is also updated in the help string, and I've removed the information for which OS the package was found as this is now no longer relevant.